ovn_k8s/watcher/watcher.py

```diff
@@ -65,14 +66,64 @@ def _process_func(watcher, watcher_recycle_func):
             watcher = watcher_recycle_func()
 
 
+def _sync_k8s_pods():
+    if variables.OVN_MODE == "overlay":
+        mode = ovn_k8s.modes.overlay.OvnNB()
+    else:
+        return
+
+    try:
+        namespaces = kubernetes.get_all_namespaces(variables.K8S_API_SERVER)
+        if namespaces and 'items' in namespaces:
+            for namespace in namespaces['items']:
+                pods = kubernetes.get_all_pods(variables.K8S_API_SERVER,
+                                               namespace['metadata']['name'])
```

Comment:
There is an api call for retrieving all pods regardless of their namespace.
Perhaps you can leverage that:

curl http://<api_server_ip>:<api_server_port>/api/v1/pods